src/main/java/com/almostreliable/merequester/wireless/WRScreen.java

```diff
@@ -0,0 +1,37 @@
+package com.almostreliable.merequester.wireless;
+
+import appeng.client.gui.style.ScreenStyle;
+import appeng.client.gui.widgets.BackgroundPanel;
+import appeng.client.gui.widgets.ToolboxPanel;
+import appeng.client.gui.widgets.UpgradesPanel;
+import appeng.menu.SlotSemantics;
+
+import com.almostreliable.merequester.client.RequesterTerminalScreen;
+
+import de.mari_023.ae2wtlib.api.terminal.IUniversalTerminalCapable;
+import de.mari_023.ae2wtlib.api.terminal.WTMenuHost;
+
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Inventory;
+
+public class WRScreen extends RequesterTerminalScreen<WRMenu> implements IUniversalTerminalCapable {
+
+    public WRScreen(WRMenu menu, Inventory playerInventory, Component name, ScreenStyle style) {
+        super(menu, playerInventory, name, style);
+        if (getMenu().isWUT())
+            addToLeftToolbar(cycleTerminalButton());
+
+        widgets.add("upgrades", new UpgradesPanel(getMenu().getSlots(SlotSemantics.UPGRADE), getMenu().getHost()));
+        if (getMenu().getToolbox().isPresent())
+            widgets.add("toolbox", new ToolboxPanel(style, getMenu().getToolbox().getName()));
+        //widgets.add("singularityBackground", new BackgroundPanel(style.getImage("singularityBackground")));
```

Comment:
this will use ae2wtlibs upgrade panels instead, which also shows the singularity (if applicable, and supports scrolling if the list of upgrades gets longer than the available space)

```suggestion
        upgradesPanel = addUpgradePanel(widgets, getMenu());
        if (getMenu().getToolbox().isPresent())
            widgets.add("toolbox", new ToolboxPanel(style, getMenu().getToolbox().getName()));
```